token/rust/src/token.rs

```diff
@@ -0,0 +1,260 @@
+use super::client::{SendTransaction, TokenClient, TokenClientError};
+use solana_sdk::{
+    instruction::Instruction,
+    program_error::ProgramError,
+    program_pack::Pack,
+    pubkey::Pubkey,
+    signer::{signers::Signers, Signer},
+    system_instruction,
+    transaction::Transaction,
+};
+use spl_associated_token_account::{create_associated_token_account, get_associated_token_address};
+use spl_token::{instruction, state};
+use std::{fmt, sync::Arc};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum TokenError {
+    #[error("client error: {0}")]
+    Client(TokenClientError),
+    #[error("program error: {0}")]
+    Program(#[from] ProgramError),
+    #[error("account not found")]
+    AccountNotFound,
+    #[error("invalid account owner")]
+    AccountInvalidOwner,
+    #[error("invalid account mint")]
+    AccountInvalidMint,
+}
+
+pub type TokenResult<T> = Result<T, TokenError>;
+
+pub struct Token<ST, TS> {
+    client: Arc<dyn TokenClient<ST>>,
+    pubkey: Pubkey,
+    payer: TS,
+}
+
+impl<ST, TS> fmt::Debug for Token<ST, TS>
+where
+    TS: Signer,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Token")
+            .field("pubkey", &self.pubkey)
+            .field("payer", &self.payer.pubkey())
+            .finish()
+    }
+}
+
+impl<ST, TS> Token<ST, TS>
+where
+    ST: SendTransaction,
+    TS: Signer,
+{
+    async fn process_ixs<T: Signers>(
+        client: &Arc<dyn TokenClient<ST>>,
+        payer: &TS,
+        instructions: &[Instruction],
+        signing_keypairs: &T,
```

Comment:
I think you can just roll `payer` into the `signing_keypairs` slice, can't you?

Reply:
I just changed `process_ixs` method, so no `client` and `payer` in arguments now.